### ISSUE TITLE
Search Results Page is Slow

### ISSUE THREAD

**Issue by href** [OneGov/onegov.town#174]
The search results page is slow if many results are returned. Even if there are not, it's still not as fast as it should be. We need to improve there.

**Comment by href** [comment]
This might have something to do with https://github.com/OneGov/onegov.town/commit/f45f5e73626dc4d0508823dc294ec3bad02782c5. It looks like the "match phrase" query we do there is quite slow. It remains to be seen if we can improve on it's speed or if we can solve this problem with [shingles](https://www.elastic.co/blog/searching-with-shingles) instead.

**Comment by href** [comment]
Turns out the problem is the fuzziness of the search. I limited the fuzziness somewhat in https://github.com/OneGov/onegov.town/commit/f5c8e8c520161d5101c60477f0ef8316f286a87f and it is quite a bit faster, without sacrificing results. The previous fuzziness value was a bit too much, as it would mostly lead to results which were a bit of a stretch.

I took this performance hunt as a chance to once again look at german compound words and I conclude again that if we want the query 'Gemeinderat' to match 'Gemeinderatsbeschluss', then we need to switch to ngrams/shingles. This approach actually beats wordlist and/or hyphenation-rule decompounding.

This would however still leat to a long tail of meaningless results. After testing this out again I remain of the opinion that not enough search results is the lesser evil.
